File: app/src/main/java/com/example/bulpyeonmarble/BoardActivity.java
package com.example.bulpyeonmarble;

import android.content.Intent;
import android.os.Bundle;
import android.widget.Button;
import android.widget.Toast;
import androidx.appcompat.app.AppCompatActivity;

public class BoardActivity extends AppCompatActivity {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_board);

        Button btnStartPasswordGame = findViewById(R.id.btnStartPasswordGame);
        Button btnStartAdGame1 = findViewById(R.id.btnStartAdGame1);
        Button btnStartAdGame2 = findViewById(R.id.btnStartAdGame2);
        Button btnStartAdGame3 = findViewById(R.id.btnStartAdGame3);
        Button btnStartDiceGame = findViewById(R.id.btnStartDiceGame);

        btnStartPasswordGame.setOnClickListener(v -> {
            startActivity(new Intent(BoardActivity.this, PasswordActivity.class));
        });

        btnStartAdGame1.setOnClickListener(v -> {
            startActivity(new Intent(BoardActivity.this, AdGame1Activity.class));
        });

        btnStartAdGame2.setOnClickListener(v -> {
            startActivity(new Intent(BoardActivity.this, AdGame2Activity.class));
        });

        btnStartAdGame3.setOnClickListener(v -> {
            startActivity(new Intent(BoardActivity.this, AdGame3Activity.class));
        });

        btnStartDiceGame.setOnClickListener(v -> {
            startActivity(new Intent(BoardActivity.this, DiceActivity.class));
        });

        int miniGameScore = getIntent().getIntExtra("miniGameScore", -1);
        if (miniGameScore != -1) {
            Toast.makeText(this, "미니게임 점수 반영: " + miniGameScore + "점", Toast.LENGTH_SHORT).show();
        }

        int diceResult = getIntent().getIntExtra("diceResult", -1);
        if (diceResult != -1) {
            Toast.makeText(this, "주사위 결과: " + diceResult + "칸 이동!", Toast.LENGTH_SHORT).show();
        }
    }
}

File: app/src/main/java/com/example/adchaosdemo/BoardActivity.kt
package com.example.adchaosdemo

import android.content.Intent
import android.os.Bundle
import android.os.CountDownTimer
import android.os.Handler
import android.os.Looper
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity

class BoardActivity : AppCompatActivity() {

    companion object {
        const val EXTRA_PLAYERS = "extra_players"
    }

    private var startCountDownTimer: CountDownTimer? = null
    private val handler = Handler(Looper.getMainLooper())
    private var moveToMiniGameRunnable: Runnable? = null
    private var lastShownCount = Int.MIN_VALUE

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_board)

        val boardCountdownText = findViewById<TextView>(R.id.boardCountdownText)
        val boardPlayersText = findViewById<TextView>(R.id.boardPlayersText)
        val players = intent.getStringArrayListExtra(EXTRA_PLAYERS).orEmpty()

        val description = buildString {
            append(getString(R.string.board_demo_title))
            append("\n\n")
            append(getString(R.string.board_participants_title))
            append("\n")
            players.forEachIndexed { index, name ->
                append("${index + 1}P  $name\n")
            }
        }
        boardPlayersText.text = description

        lastShownCount = 5
        boardCountdownText.text = getString(R.string.board_start_countdown, lastShownCount)
        startCountDownTimer = object : CountDownTimer(5000L, 1000L) {
            override fun onTick(millisUntilFinished: Long) {
                val remainingSeconds = ((millisUntilFinished + 999L) / 1000L).toInt()
                if (remainingSeconds in 1..4 && remainingSeconds != lastShownCount) {
                    lastShownCount = remainingSeconds
                    boardCountdownText.text = getString(R.string.board_start_countdown, remainingSeconds)
                }
            }

            override fun onFinish() {
                boardCountdownText.text = getString(R.string.board_start_now)
                moveToMiniGameRunnable = Runnable {
                    startActivity(Intent(this@BoardActivity, MiniGame3IntroActivity::class.java))
                    finish()
                }.also { handler.postDelayed(it, 1500L) }
            }
        }.start()
    }

    override fun onDestroy() {
        startCountDownTimer?.cancel()
        startCountDownTimer = null
        lastShownCount = Int.MIN_VALUE
        moveToMiniGameRunnable?.let { handler.removeCallbacks(it) }
        moveToMiniGameRunnable = null
        super.onDestroy()
    }
}
